### ISSUE TITLE
uBlock breaks sites using mParticle

### ISSUE THREAD

**Issue by tisawin** [uBlockOrigin/uAssets#11562]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

Any site that uses mParticle for marketing tracking and setting user identity  breaks for users with uBlock.  

For example:
https://www.gymshark.com/ (You can see it's being blocked by client

https://trends.builtwith.com/websitelist/mParticle
^Here are other sites that uses mParticle

### Describe the issue

For example, our particular site uses it to set user identity to create tracking for that user.  uBlock will block calls to mParticle, which breaks some features on our site.  We'd like mParticle to be white-listed if possible.

https://docs.mparticle.com/
^mParticle library we're using

### Screenshot(s)

![Screenshot 2022-02-02 213420](https://user-images.githubusercontent.com/16124260/152272066-fac6d50d-36c6-4869-a7a0-1f21edf03752.png)

### uBlock Origin version

uBlock Origin 1.40.8

### Browser name and version

Chrome Version 98.0.4758.8

### Settings

n/a  Just installed

### Notes

FYI, this also breaks Sentry, Optimizely, & New Relic, which are common dev tools!

**Comment by uBlock-user** [comment]
> breaks for users with uBlock

What exactly is broken ?

> We'd like mParticle to be white-listed if possible.

There's no whitelist in uBO.

**Comment by tisawin** [comment]
@uBlock-user 
We use branch-mparticle-web-kit on our siite.
We set window.branch.setIdentity() after a user fills our login forms to be able to login.  We also use this feature to be able to set the user and set attributes.  We also have hundred of thousands of users - not too sure how many have uBlock installed.  We just know that uBlock blocks it after doing some testing.

Is there a more appropriate place to put this issue?  It would be great to be pointed to the right direction.

**Comment by uBlock-user** [comment]
>  We use branch-mparticle-web-kit on our siite.

Which site would that be ?

**Comment by tisawin** [comment]
KFC.com

**Comment by okiehsch** [comment]
> Any site that uses mParticle for marketing tracking and setting user identity breaks for users with uBlock.

How does it "break"?

**Comment by okiehsch** [comment]
Let's use `https://www.gymshark.com/` as an example, I have an account, what can't I do?

**Comment by tisawin** [comment]
Try logging into our site, kfc.com with uBlock on. :')

**Comment by okiehsch** [comment]
I can't test that site, that's why I used the first site you mentioned.

**Comment by tisawin** [comment]
KFC.com?  Ahhh, are you based outside of US?

**Comment by okiehsch** [comment]
Yes.

**Comment by tisawin** [comment]
Would you like me to send a video/more screenshots? 
Another choice (you don't have to) is going on vpn if you have it.  

Let me know if this is too much trouble too.  I don't mind closing this thread out in that case.

**Comment by okiehsch** [comment]
>  Another choice (you don't have to) is going on vpn if you have it.

I am well aware of that fact. 😉 
I don't have an account for kfc, I do have an account for gymshark.
If users can't login on a specific site we will add an exception for that site, what we won't do is whitelist a tracking solution globally.

**Comment by tisawin** [comment]
I'm not sure how gymshark is using mparticle, but you can see that calls get blocked for them.

We're using it to set the customer profile, email subscription, etc (pretty iffy utilization of mparticle imo but I joined the project way after the fact).

**Comment by uBlock-user** [comment]
> We're blocked from setting user attributes using mparticle

So you're unable to do any tracking because filters are in place to block just that. Don't see how users are affected.

**Comment by okiehsch** [comment]
> but you can see that calls get blocked for them.

Yes, but I can use the site without issue as far as I can see, the email subscription works, I can by stuff, etc.

**Comment by tisawin** [comment]
> > but you can see that calls get blocked for them.
> 
> Yes, but I can use the site without issue as far as I can see, the email subscription works, I can by stuff, etc.

If you're referring to gymshark, it's because they're using mparticle as truly a marketing tool unlike my project I'm working on.  

Just let me know if anything could be done or proof is needed haha!  I can just close this out if nothing can be done.

**Comment by okiehsch** [comment]
>   mparticle to store if user is subscribed or unsubscribed to our emails ..... unsubscribed through our site, we currently can't fully unsubscribe them

And you use mparticle for that, I agree that is an unusal utilization.
Anyway, I don't think anything can be done on our end, mparticle will remain globally blocked.

**Comment by guinguinguinguin** [comment]
Cool cool - thanks!

**Comment by iam-py-test** [comment]
> For example, our particular site uses it to set user identity to create tracking for that user.

And you want it to be unblocked?

**Comment by gwarser** [comment]
This is something similar to Google Tag Manager where you can run any crap on user machines through unified API?

**Comment by guinguinguinguin** [comment]
I'm not sure if you're sincerely asking or being sarcastic. 😅

**Comment by iam-py-test** [comment]
Unless it is preventing the user from using the website, then I don't think we plan on unblocking it
>  Either way, it also doesn't really answer why other monitoring tools are being blocked as well, like New Relic or Sentry etc.

Decision by @easylist not uBo. I personally feel that those are a privacy risk (and besides, users still can report issues & provide their own data on their own terms) but I see why some people think they are needed
